bundle/edu.gemini.obslog/src/main/scala/edu/gemini/obslog/instruments/GNIRSLogSegment.scala

```diff
@@ -0,0 +1,154 @@
+      if (yesOrNotNo(c)) "XD".some
+      else map.sgetOpt(WavelengthKey).map { l => \/.fromTryCatchNonFatal {
+          val lambda = l.toDouble
+          FilterOrder.find(_.getMaxWavelength > lambda).getOrElse(GNIRSParams.Order.ONE).getBand
+        }.getOrElse("?") }
+      }
+
+    // If we can determine the filter in one of the two ways, then adjust it.
+    determineFilter.orElse(determineFilterFromWavelength).foreach(map.put(FilterKey, _))
+  }
+
+  // Replace the disperser value with disperser/wavelength.
+  private def decorateDisperser(map: ConfigMap): Unit = for {
+    v <- map.sgetOpt(DisperserKey)
+    d <- Option(Disperser.getDisperser(v)).map(_.logValue())
+    w <- map.sgetOpt(WavelengthKey)
+  } map.put(DisperserKey, s"$d/$w")
+
+  // This method looks at the coadds, readmode, and exposureTime to synthesize the new complicated exposure time
+  private def decorateExposureTime(map: ConfigMap): Unit = {
+    def lowNoiseReads: Option[Int] = {
+      val lnrs = map.sgetOpt(ReadModeKey).map(ReadMode.getReadMode).map(_.getLowNoiseReads)
+      if (lnrs.isEmpty) Log.severe("No readMode found in GNIRS items")
+      lnrs
+    }
+    def extractWithLog(key: String, err: String, default: String): String = {
+      val value = map.sgetOpt(key)
+      if (value.isEmpty) Log.severe(err)
+      value.getOrElse(default)
+    }
+
+    lowNoiseReads.foreach { lnrs =>
+      val expTime = extractWithLog(ExposureTimeKey, "No exposureTime property in GNIRS items", "1")
+      val coadds = extractWithLog(CoaddsKey, "No coaddsValue found in GNIRS items", "1")
+
+      map.put(ExposureTimeKey, s"$expTime/$lnrs/$coadds")
+    }
+  }
+
+  // Replace the slitwidth with the log value.
+  private def decorateSlitWidth(map: ConfigMap): Unit = for {
+    w <- map.sgetOpt(SlitWidthKey)
+    s <- Option(SlitWidth.getSlitWidth(w))
+  } map.put(SlitWidthKey, s.logValue())
+
+  // This changes the acquisition mirror from "in" to "Y"; otherwise nothing.
+  private def decorateAcquisitionMirror(map: ConfigMap): Unit = {
+    val newValue = map.sgetOpt(AcqMirrorKey).collect {
+      case "in" => "Y"
+    }.getOrElse("")
+    map.put(AcqMirrorKey, newValue)
+  }
+
+  private def decorateCamera(map: ConfigMap): Unit = {
+    for {
+      k <- map.sgetOpt(WavelengthKey)
+      p <- map.sgetOpt(PixelScaleKey)
+      c <- map.sgetOpt(CrossDispersedKey)
+    } yield {
+      val ps = PixelScale.getPixelScale(p)
+
+      // Camera
+      val cameraName = \/.fromTryCatchNonFatal {
+        val lambda = k.toDouble
+        ps match {
+          case PixelScale.PS_015 if lambda > 2.5 => Camera.SHORT_RED
+          case PixelScale.PS_015                 => Camera.SHORT_BLUE
+          case _                 if lambda > 2.5 => Camera.LONG_RED
+          case _                                 => Camera.LONG_BLUE
+        }
+      }.getOrElse("?")
+
+      // Prism
+      val prismName = yesOrNotNo(c) ? (ps match {
```

Comment:
This is a difference from the previous version, and also the bug fix we're trying to do?  Now it is "if not no" instead of "if yes".